modules/api/src/main/java/org/apache/fluo/api/observer/ObserversFactory.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.fluo.api.observer;
+
+import java.util.function.BiConsumer;
+
+import org.apache.fluo.api.client.FluoClient;
+import org.apache.fluo.api.config.FluoConfiguration;
+import org.apache.fluo.api.config.SimpleConfiguration;
+import org.apache.fluo.api.data.Bytes;
+import org.apache.fluo.api.data.Column;
+import org.apache.fluo.api.metrics.MetricsReporter;
+import org.apache.fluo.api.observer.Observer.NotificationType;
+
+/**
+ * Fluo Workers use this class to create {@link Observer}s to process notifications. Implementations
+ * of this class should emit zero or more {@link Observer}s.
+ *
+ * <p>
+ * When Fluo is initialized {@link #getObservedColumns(Context, BiConsumer)} is called. The columns
+ * it emits are stored in Zookeeper. Transactions will use the columns stored in Zookeeper to
+ * determine when to set notifications. When Workers call
+ * {@link #createObservers(ObserverConsumer, Context)}, the columns emitted must be the same as
+ * those emitted during initialization. If this is not the case, then the worker will fail to start.
+ *
+ * @see FluoConfiguration#setObserversFactory(String)
+ * @since 1.1.0
+ */
+public interface ObserversFactory {
```

Comment:
The pluralization of the name is a bit confusing. `ObserverFactory` vs. `ObserversFactory`. Also, I think this naming convention works best if there's a getter method which returns the object named by the part before the word "Factory". So, this name looks to me like it should have a `Observers get()` method, or a `Collection<Observer> get()` method. I'm not sure I have a better name, but perhaps if the "getter" looked like `Set<Observer> get(Context)`, it would be slightly more intuitive (in my mind, anyway). At the very least, I think it's weird to have a Factory interface which doesn't return anything.